### PR TITLE
TODO3

### PR DESCRIPTION
**Pull request by simkasss**

### CODE REVIEW

File: contracts/FundMe.sol [hunk 1]
```diff
@@ -71,6 +71,8 @@ contract FundMe {
         ) {
             address funder = funders[funderIndex];
             addressToAmountFunded[funder] = 0; //3TODO: why is this necessary. How is this pattern called.
+            // 3ANS: This is Checks-Effects-Interactions pattern, it is used to avoid re-entrancy.
+            // First we have to change the values of Funded Amounts and just then we can call a transaction
```
Comment (DeividasK) [inline contracts/FundMe.sol]:
```suggestion
            // First we change the values of Funded Amounts and then we call a transaction
```
You don't "have to" change the values.

File: contracts/FundMe.sol [hunk 2]
```diff
@@ -71,6 +71,8 @@ contract FundMe {
         ) {
             address funder = funders[funderIndex];
             addressToAmountFunded[funder] = 0; //3TODO: why is this necessary. How is this pattern called.
+            // 3ANS: This is Checks-Effects-Interactions pattern, it is used to avoid re-entrancy.
```
Comment (DeividasK) [inline contracts/FundMe.sol:74]: What is re-entrancy? How does following this pattern protect from it?

**Review by DeividasK — CHANGES_REQUESTED** [review]

File: contracts/FundMe.sol (hunk 2 above)
Comment (simkasss) [reply contracts/FundMe.sol:74]: I will try to describe it in my own words. 
Re-entrancy is when a user calls the function for a second time before the function was executed.
If we change the amount value to zero after the function call, when a function is called the value will not be changed until the function will be executed. In this case, while the function execution is not over and the amount value is not changed, a user could call a function for a second time.
When using this pattern, amount value would already be zero before the execution of the function starts, in this way the function could not be called during the execution of the function.

Comment (DeividasK) [reply contracts/FundMe.sol:74]: Created https://github.com/simkasss/hardhat-fund-me/issues/5 for a practical exploration task.

**Review by DeividasK — APPROVED** [review]

**Pull request merged**